modules/AlphaWalletFoundation/AlphaWalletFoundation/SwapToken/NativeSwap/Helpers/ApproveSwapProvider.swift

```diff
@@ -83,28 +91,15 @@ public final class ApproveSwapProvider {
             return .fail(SwapError.unknownError)
         }
 
-        let provider = WaitTillTransactionCompleted(blockchainProvider: configurator.session.blockchainProvider)
-
         return delegate
             .promptForErc20Approval(token: token, server: server, owner: owner, spender: spender, amount: amount, in: self)
             .mapError { SwapError.inner($0) }
-            .flatMap { [provider] transactionId -> AnyPublisher<Bool, SwapError> in
+            .flatMap { [provider] hash -> AnyPublisher<Bool, SwapError> in
                 self.delegate?.changeState(in: self, state: .waitTillApproveCompleted)
 
-                return provider.waitTillCompleted(hash: transactionId)
+                return provider.transaction(hash: hash, for: .completed, timeout: 3600)
                     .map { _ in return true }
-                    .catch { error -> AnyPublisher<Bool, SwapError> in
-                        if error.embedded is WaitTillTransactionCompleted.NotCompletedYetError {
-                            return .fail(SwapError.approveTransactionNotCompleted)
-                        } else if let error = error.embedded as? SwapError {
-                            switch error {
-                            case .userCancelledApproval:
-                                return .just(false)
-                            case .unableToBuildSwapUnsignedTransactionFromSwapProvider, .invalidJson, .unableToBuildSwapUnsignedTransaction, .approveTransactionNotCompleted, .unknownError, .tokenOrSwapQuoteNotFound, .inner:
-                                return .fail(error)
```

Comment:
Removing this means we no longer display the error, right? Did we display the error previously?

Reply:
nope, that is just for waiting pending transaction, we didn't display error for it even before, the error that could be generated throws on top level, so we still able to display it